pkg/configs/db/postgres/postgres.go

```diff
@@ -134,6 +135,86 @@ func (d DB) GetConfigs(since configs.ID) (map[string]configs.View, error) {
 	})
 }
 
+// GetRulesConfig gets the latest alertmanager config for a user.
+func (d DB) GetRulesConfig(userID string) (configs.VersionedRulesConfig, error) {
+	current, err := d.GetConfig(userID)
+	if err != nil {
+		return configs.VersionedRulesConfig{}, err
+	}
+	cfg := current.GetVersionedRulesConfig()
+	if cfg == nil {
+		return configs.VersionedRulesConfig{}, sql.ErrNoRows
+	}
+	return *cfg, nil
+}
+
+// SetRulesConfig sets the current alertmanager config for a user.
+func (d DB) SetRulesConfig(userID string, oldConfig, newConfig configs.RulesConfig) (bool, error) {
+	updated := false
+	err := d.Transaction(func(tx DB) error {
+		current, err := d.GetConfig(userID)
+		if err != nil && err != sql.ErrNoRows {
+			return err
+		}
+		// The supplied oldConfig must match the current config. If no config
+		// exists, then oldConfig must be nil. Otherwise, it must exactly
+		// equal the existing config.
+		if !((err == sql.ErrNoRows && oldConfig == nil) || reflect.DeepEqual(current.Config.RulesFiles, oldConfig)) {
+			return nil
+		}
+		new := configs.Config{
+			AlertmanagerConfig: current.Config.AlertmanagerConfig,
+			RulesFiles:         newConfig,
+		}
+		updated = true
+		return d.SetConfig(userID, new)
+	})
+	return updated, err
+}
+
+func (d DB) findRulesConfigs(filter squirrel.Sqlizer) (map[string]configs.VersionedRulesConfig, error) {
+	rows, err := d.Select("id", "owner_id", "config ->> 'rules_files'").
+		Options("DISTINCT ON (owner_id)").
+		From("configs").
+		Where(filter).
+		Where("config ->> 'rules_files' <> '{}'").
+		OrderBy("owner_id, id DESC").
```

Comment:
Why don't we have a `Limit(1)` here like in `GetConfig()` ?

Reply:
Because `GetConfig()` gets the configuration for a single user, whereas `findRulesConfigs` helps `GetAllRulesConfigs()` and `GetRulesConfigs()` get the active configurations for all users. 

I've added a doc comment that highlights the purpose, but doesn't explicitly answer the "why no `Limit(1)`?" question:

> // findRulesConfigs helps GetAllRulesConfigs and GetRulesConfigs retrieve the
> // set of all active rules configurations across all our users.

I'm open to renaming some or all of these functions if you can suggest names that more clearly communicate intent.